File: golf.py
score_dict = {"albatross" : -3 , "eagle" : -2 , "birdie" : -1  , "bogey" : 1 , "double-bogey" : 2 , "triple-bogey" : 3,"par" : 0}

scores_description = {1:"1 over par" , 2 : "2 over par" , 3:"3 over par"  , -2 : "-2 under par" , -1 : "1 under par" , 0 : "exact no of strokes specified" ,  }

input_list = ["eagle","bogey","par","bogey","double-bogey","birdie","bogey","par","birdie","par","par","par","par","par","bogey","eagle","bogey","par"]

score = [(score_dict[input_list[i]]) for i in range(18)]

print(sum(score))

scores_description[sum(score)]

File: golf.hs
import Data.Map

golfterms_dict = [("par", 0), ("albatross", -3), ("eagle", -2), ("birdie", -1), ("bogey", 1), ("double-bogey", 2), ("triple-bogey", 3)]
    input_list = ["eagle", "bogey", "par", "bogey", "double-bogey", "birdie", "bogey", "albatross", "birdie", "par", "par", "par", "par", "par","bogey", "eagle", "bogey", "par"]

score ::[String] -> Integer
score s = sum [fromList golfterms_dict !x | x <- s]

scoreInPar :: Integer -> String
scoreInPar n | n == 0 =  "par"
             | n > 0  = show n ++ " over par"
             | n < 0  = show (abs n) ++ " under par"

main = do
print $ scoreInPar $ score input_list
